src/parser.cpp

```diff
@@ -1125,4 +1157,216 @@ void Parser::ParserImpl::loadImport(const ImportSourcePtr &importSource, const M
+            if (referencedVariable == nullptr) {
+                ErrorPtr err = std::make_shared<Error>();
+                err->setDescription("Reset referencing variable '" + variableReference +
+                                    "' is not a valid reference for a variable in component '" + component->getName() + "'.");
+                err->setReset(reset);
+                err->setKind(Error::Kind::RESET);
+                err->setRule(SpecificationRule::RESET_INVALID_VARIABLE_REFERENCE);
+                mParser->addError(err);
+            } else {
+                reset->setVariable(referencedVariable);
+            }
+        } else if (attribute->isType("order")) {
+            if (convertToInt(attribute->getValue(), &order)) {
+                orderDefined = true;
+            } else {
+                if (reset->getVariable() != nullptr) {
+                    variableName = reset->getVariable()->getName();
+                }
+                ErrorPtr err = std::make_shared<Error>();
+                err->setDescription("Reset in component '" + component->getName() +
+                                    "' referencing variable '" + variableName +
+                                    "' has a non-integer order value '" + attribute->getValue() + "'.");
+                err->setReset(reset);
+                err->setKind(Error::Kind::RESET);
+                err->setRule(SpecificationRule::RESET_NON_INT_ORDER);
+                mParser->addError(err);
+            }
+        } else if (attribute->isType("id")) {
+            reset->setId(attribute->getValue());
+        } else {
+            ErrorPtr err = std::make_shared<Error>();
+            err->setDescription("Reset in component '" + component->getName() +
+                                "' has an invalid attribute '" + attribute->getType() + "'.");
+            err->setReset(reset);
+            err->setKind(Error::Kind::RESET);
+            mParser->addError(err);
+        }
+        attribute = attribute->getNext();
+    }
+
+    if (referencedVariable == nullptr) {
+        ErrorPtr err = std::make_shared<Error>();
+        err->setDescription("Reset in component '" + component->getName() +
+                            "' does not reference a variable in the component.");
+        err->setReset(reset);
+        err->setKind(Error::Kind::RESET);
+        mParser->addError(err);
+    }
+
+    if (reset->getVariable() != nullptr) {
+        variableName = reset->getVariable()->getName();
+    }
+    if (orderDefined) {
+        reset->setOrder(order);
+    } else {
+        ErrorPtr err = std::make_shared<Error>();
+        err->setDescription("Reset in component '" + component->getName() +
+                            "' referencing variable '" + variableName +
+                            "' does not have an order defined.");
+        err->setReset(reset);
+        err->setKind(Error::Kind::RESET);
+        mParser->addError(err);
```

Comment:
Am I correct in assuming that we will get two error messages if a non-integer value is provided for the order? If so, is that really what we want?

Reply:
Yes, I think it is what we want as we are actually violating two rules.

Reply:
Ok, fair enough, I don't (yet) feel too strongly about it.